src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -453,12 +462,12 @@ private ClusterDescription ProcessReplicaSetChange(ClusterDescription clusterDes
                                         args.NewServerDescription.EndPoint,
                                         args.NewServerDescription.ReplicaSetConfig.Version,
                                         args.NewServerDescription.ElectionId)));
+
+                                _maxElectionInfo = new ElectionInfo(
+                                    args.NewServerDescription.ReplicaSetConfig.Version.Value,
+                                    args.NewServerDescription.ElectionId);
                             }
                         }
-
```

Comment:
Was this moved so that the logic to update the _maxElectionInfo would be closer to where the SDAM monitoring events were being generated?

Reply:
I moved it to account for the change here: https://github.com/MikalaiMazurenka/mongo-csharp-driver/pull/37/files/b1b950bbf1f5f2add235f821eceb0b194916cd27#diff-346021f0b586b815ec28dc10028a0a5aR447
It used to be `args.NewServerDescription.ElectionId` for every code path, but now one of code paths has `args.NewServerDescription.ElectionId ?? _maxElectionInfo.ElectionId` instead.